jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2282,45 +2282,83 @@ int toFlag() {
 			}
+
 			ClassDefiner(String name, byte[] template, Lookup lookupObj) {
 				className = name;
 				classBytes = template;
 				lookup = lookupObj;
 				ClassOptFlags = 0;
 			}
+
 			Class<?> defineClass(boolean initOption) {
 				Class<?> ret = AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
 					@Override
 					public Class<?> run() {
 						JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
 						Class<?> lookupClass = lookup.lookupClass();
-						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes, jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, null);
+						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes,
+								jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, null);
+					}
+				});
+				return ret;
+			}
+
+			Class<?> defineClass(boolean initOption, Object classData) {
+				Class<?> ret = AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
+					@Override
+					public Class<?> run() {
+						JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
+						Class<?> lookupClass = lookup.lookupClass();
+						return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes,
+								jlAccess.protectionDomain(lookupClass), initOption, ClassOptFlags, classData);
 					}
 				});
 				return ret;
 			}
 		}
+
 		/**
 		 * Constructs a new hidden class from an array of class data bytes.
 		 * 
-		 * @param bytes the class data bytes of the hidden class to be defined.  
+		 * @param bytes the class data bytes of the hidden class to be defined.
 		 * @param initOption whether to initialize the hidden class.
 		 * @param classOptions the {@link ClassOption} to define the hidden class.
+		 * 
 		 * @return A Lookup object of the newly created hidden class.
 		 * @throws IllegalAccessException if this Lookup does not have full privilege access.
 		 */
-
 		public Lookup defineHiddenClass(byte[] bytes, boolean initOption, ClassOption... classOptions) throws IllegalAccessException {
+			ClassDefiner definer = classDefiner(bytes, classOptions);
+			return new Lookup(definer.defineClass(initOption));
+		}
+		
+		/**
+		 * Constructs a new hidden class from an array of class data bytes.
+		 * 
+		 * @param bytes the class data bytes of the hidden class to be defined.
+		 * @param classData the classData to be stored in the hidden class.
+		 * @param classOptions the {@link ClassOption} to define the hidden class.
+		 * 
+		 * @return A Lookup object of the newly created hidden class.
+		 * @throws IllegalAccessException if this Lookup does not have full privilege access.
+		 */
+		Lookup defineHiddenClassWithClassData(byte[] bytes, Object classData, ClassOption... classOptions) throws IllegalAccessException {
+			ClassDefiner definer = classDefiner(bytes, classOptions);
+			return new Lookup(definer.defineClass(true, classData));
```

Comment:
How did we determine that `true` aka initialize is the right value here?

Reply:
From the [documentation](https://github.com/ibmruntimes/openj9-openjdk-jdk15/blob/88d915c8537bf7e6a9828c0523cb7478cf20ed10/src/java.base/share/classes/java/lang/invoke/MethodHandles.java#L1973-L1974). `defineHiddenClassWithClassData` is a package private method so its documentation does not get published in the official docs.